Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h

```diff
@@ -175,19 +192,40 @@ std::unique_ptr<T> create(const IndexArg &indexArg, const HistArg &histArg) {
   return std::move(ws);
 }
 
+template <class T, class IndexArg,
+          typename std::enable_if<
+              !std::is_base_of<API::MatrixWorkspace, IndexArg>::value>::type * =
+              nullptr>
+std::unique_ptr<T> create(const IndexArg &indexArg,
+                          const HistogramData::Histogram &histogram) {
+  auto ws = Kernel::make_unique<T>();
+  HistogramData::Histogram histogramTemplate(histogram);
+  if (std::is_base_of<DataObjects::EventWorkspace, T>::value) {
+    histogramTemplate = detail::stripData(histogramTemplate);
+  }
+  ws->initialize(indexArg, histogramTemplate);
+  return std::move(ws);
+}
+
 template <class T, class P,
           typename std::enable_if<std::is_base_of<API::MatrixWorkspace,
                                                   P>::value>::type * = nullptr>
 std::unique_ptr<T> create(const P &parent) {
-  return create<T>(parent, parent.getNumberHistograms(),
-                   detail::stripData(parent.histogram(0)));
-}
+  const auto numHistograms = parent.getNumberHistograms();
+  if (parent.isCommonBins()) {
+    return create<T>(parent, numHistograms,
+                     detail::stripData(parent.histogram(0)));
+  }
+  std::unique_ptr<T> ws = detail::createUninitialized<T, P>(parent);
+  const auto YLength = parent.blocksize();
+  const auto XLength = parent.isHistogramData() ? YLength + 1 : YLength;
+  ws->initialize(numHistograms, XLength, YLength);
```

Comment:
I think you can simply call the `create(parent, numHistograms, detail::stripData(parent.histogram(0)));` method here, and then set the correct X afterwards, the result is identical. As a result, you do not need to split out the `createUninitialized` method from the original `create` method. There should be no performance penalty to this.